### ISSUE TITLE
Search should listen to `input` event instead of `keyup`

### ISSUE THREAD

**Issue by denilsonsa** [javve/list.js#363]
[This line][1]:

    events.bind(getByClass(list.listContainer, list.searchClass), 'keyup', function(e) {

Should use `'input'` event.

Why? I can give you a few reasons:

* The `input` event will not fire if the user presses keys but doesn't change the content (e.g. arrow keys, tab, shift).
* The `input` event will fire if the input field is modified by means other than keyboard (e.g. drag-and-dropping a selection onto the text field, right-clicking and pasting or deleting, middle-clicking to paste).
* [Browser support is great][2] (IE9+, and essentially any other browser).

[1]: https://github.com/javve/list.js/blob/8c1c6fd12862d0217198a3d6ff2d5b79f7b78c52/src/search.js#L101

[2]: https://developer.mozilla.org/en-US/docs/Web/API/GlobalEventHandlers/oninput

**Comment by denilsonsa** [comment]
Actually, sorry, I just noticed you already use `input` a few lines down. But you only use it to detect clearing the field, and not for actual changes. I'd suggest removing `keyup` and only using `input`.

**Comment by wolffe** [comment]
I agree. I'm working with a search system based on clicks and simulate ENTER keypress and it's not working properly.

**Comment by javve** [comment]
Hmm, according http://caniuse.com/#feat=input-event the `input` event doesn't fire when backspace is used in IE9 :disappointed: 

Maybe I can add some browser sniffing here...

**Comment by r8n5n** [comment]
Adding support for 'focusout' would be useful here too. I had an issue where I was clearing the input field value when it lost focus but the list was keeping the search.

I copied the 'keyup' function and used it with a 'focusout' event to resolve this.
